### PR TITLE
Adding tests for CLI, factory and suite

### PR DESCRIPTION
**Pull request by aneesh404**
Added and modified tests for cli interface, benchmark factory and benchmark suite.

### CODE REVIEW

File: tests/test_benchmark_factory.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import unittest
+import os
+import sys
+
+sys.path.append(os.path.dirname(os.path.dirname(__file__)))
+from common.benchmark_factory import BenchmarkFactory
+from common.benchmark_wrapper import BenchmarkWrapper
+
+
+class PermuteArguments(unittest.TestCase):
```
Comment (gabrielefronze) [inline tests/test_benchmark_factory.py:10]: I don't really understand the class name! I Would go with some "TestArguments" instead of "PermuteArguments".

**Review by gabrielefronze — CHANGES_REQUESTED** [review]
Just a minor name change before merging! Thanks

File: tests/test_benchmark_factory.py (hunk 1 above)
Comment (aneesh404) [reply tests/test_benchmark_factory.py]: The above change has been made

**Comment by leggerf** [comment]
looks good to me. @gabrielefronze pls merge if ok for you

**Pull request merged**